internal/changestream/stream/stream.go

```diff
@@ -456,7 +456,7 @@ const (
 	// If the namespace is later deleted, you'll no longer locate that during
 	// a select.
 	selectQuery = `
-SELECT MAX(c.id), c.edit_type_id, n.namespace, changed, created_at
+SELECT MAX(c.id), c.edit_type_id, n.namespace, changed, discriminator, created_at
 	FROM change_log c
 		JOIN change_log_edit_type t ON c.edit_type_id = t.id
 		JOIN change_log_namespace n ON c.namespace_id = n.id
```

Comment:
All events are coalesced at the database level via the use of the GROUP BY in this query. If the discriminator is not the same for the namespace and the changed value throughout the usage of the event then the discriminator will not be preserved (FWW). In other words, the discriminator MUST be a constant.

Reply:
Yes, it is. It is invariant. eg units have an application_uuid and it never changes once set.